src/quacc/atoms/skzcam.py

```diff
@@ -1,73 +1,75 @@
 from __future__ import annotations
 
 from pathlib import Path
-from typing import Dict, Union
+from typing import TYPE_CHECKING
 
 import numpy as np
-from ase import Atoms, io
+from ase import Atoms
+from ase.io import write
 from ase.units import Bohr
 
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
 
-def convert_pun_to_Atoms(
-    pun_filename: Union[str, Path], atom_oxi_states: Dict
+def convert_pun_to_atoms(
+    pun_file: str | Path, atom_oxi_states: dict[str, float]
 ) -> Atoms:
     """
     Reads a .pun file and returns an ASE Atoms object containing the atomic coordinates, point charges/oxidation states and atom types.
 
     Parameters
     ----------
-    pun_filename : str
+    pun_file
```

Comment:
FYI: I used `pun_file` instead of `pun_filename` since it's technically a file path and not the name itself.